### DISCUSSION TITLE
plex hardware acceleration via intel arc a310

### DISCUSSION THREAD

**Discussion by lax556** [tteck/Proxmox#3162] [Help Needed] [answered]
Trying to get plex setup correctly using your script and an arc a310 gpu. I have gotten as far as seeing the device in the plex hardware acceleration options drop down, but when I try and transcode, it is using the cpu and not gpu ( (hw) not showing in transcode . I have been trying to get this working for a few days with no success, any help would be appreciated!  

I noticed that my /dev/dri is using card 1 and not 0, so i added it into the /pve/lxc/xxxx.conf

```
root@pve:~# ls -lha /dev/dri/

total 0
drwxr-xr-x  3 root root        120 Jun 10 09:30 .
drwxr-xr-x 19 root root       4.7K Jun 10 09:30 ..
drwxr-xr-x  2 root root         80 Jun 10 09:29 by-path
----------  1 root root          0 Jun 10 09:30 card0
crw-rw----  1 root video  226,   1 Jun 10 09:29 card1
crw-rw----  1 root render 226, 128 Jun 10 09:29 renderD128
```

```
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:1 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file
```

I can also see that plex is seeing the GPU 

```
Jun 10, 2024 09:30:13.377 [139212168899384] INFO - Preemptively preparing driver imd for GPU DG2 [Arc A310]
Jun 10, 2024 09:30:13.378 [139212168899384] INFO - Preemptively preparing driver icr for GPU DG2 [Arc A310]
Jun 10, 2024 09:33:02.195 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.223 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.233 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.239 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.246 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.253 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.259 [139212122762040] DEBUG - [Req#135/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.328 [139212118567736] INFO - [Req#138/Transcode] Preparing driver icr for GPU DG2 [Arc A310]
Jun 10, 2024 09:33:02.328 [139212118567736] DEBUG - [Req#138/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (DG2 [Arc A310])
Jun 10, 2024 09:33:02.331 [139212118567736] INFO - [Req#138/Transcode] Preparing driver imd for GPU DG2 [Arc A310]
```

**Comment by tteck** [comment]
Are you a Plex Pass member?

**Reply by lax556** [reply]
Yes I am :)
![image_2024-06-10_101320803](https://github.com/tteck/Proxmox/assets/112194588/8701cfa5-6430-4799-878e-4e9384fbed69)

**Reply by lax556** [reply]
here is my plex transcoder settings
![image_2024-06-10_101420732](https://github.com/tteck/Proxmox/assets/112194588/126fd403-babc-470f-a610-b02f2e298361)

**Reply by tteck** [reply, marked as answer]
Try disabling HDR tone mapping

**Reply by lax556** [reply]
That worked! is there a reason why?

**Reply by tteck** [reply]
I think kernel issues. I heard Plex is supposed to be coming out with a fix.

**Reply by lax556** [reply]
well 2 days of troubleshooting and its not my fault! hahaha, well I'll keep an eye out for the update. Thanks for you help and almost instant reply! Love your proxmox scripts

**Reply by rc5kamps** [reply]
> well 2 days of troubleshooting and its not my fault! hahaha, well I'll keep an eye out for the update. Thanks for you help and almost instant reply! Love your proxmox scripts

In the mean time - I found that downloaded Proxmox's Kernel to 6.5 works with Intel Tone Mapping - I am running Plex on a LXC with a i9 14900K with working HDR and full Transcoding.   If interested, here are instructions on how to do that:

TO DOWNGRADE Kernel to support DKMS/vGPU (Plex)
	Ref: https://www.derekseaman.com/2023/11/proxmox-ve-8-1-windows-11-vgpu-vt-d-passthrough-with-intel-alder-lake.html
	1. apt update
	2. apt install proxmox-headers-6.5.13-3-pve
	3. apt install proxmox-kernel-6.5.13-3-pve-signed
	4. proxmox-boot-tool kernel pin 6.5.13-3-pve
	5. proxmox-boot-tool refresh
	6. reboot
	7. pveversion  - After reboot verify your kernel is 6.5.13-3
